control/control.go

```diff
@@ -345,6 +345,10 @@ func (c *Controller) Solve(ctx context.Context, req *controlapi.SolveRequest) (*
 		procs = append(procs, proc.ForceRefsProcessor, proc.SBOMProcessor(ref.String()))
 	}
 
+	if attrs, ok := attests["provenance"]; ok {
+		procs = append(procs, proc.ForceRefsProcessor, proc.ProvenanceProcessor(attrs))
+	}
```

Comment:
We can avoid appending two copies of `ForceRefsProcessor` - I think we should add it if `len(attests) > 0` above, and then we can avoid adding it in each if branch separately.

Reply:
What do we do with unknown attests?

Reply:
I think we should error - if a user requests attestations of type `foo` and we cannot provide them, then we shouldn't build an image without them.

This would be important if/when we add more attestation types, older servers shouldn't fail to produce attestations that the user requests.